src/main/scala/piconot/internal/API.scala

```diff
@@ -0,0 +1,139 @@
+package piconot.internal
+
+import picolib.semantics._
+import scala.collection.mutable.MutableList
+
+//import scala.language.implicitConversions
+import picolib.maze.Maze
+import scalafx.application.JFXApp
+
+class API(val mazeFilename: String) extends JFXApp {
+  
+  /**
+   * List of rules created by using the API
+   */
+  val rules = MutableList.empty[RuleStructure]
+  
+  // string of possible unicode arrow characters "↑→↓←"
+  
+  /**
+   * Case class for containing the block and open strings
+   */
+  case class BlockOpen(block: String, open: String)
+  
+  /**
+   * Case class for containing the move and toState strings
+   */
+  case class MoveToState(move: String, toState: String)
+  
+  /**
+   * Class for constructing and converting rules
+   */
+  class RuleStructure(stateName: String) {    
```

Comment:
The layout of your internal DSL code in classes and case classes seems to generally make more sense than ours - we essentially had a file for each syntactic element, which we felt was out of necessity. I think that's mostly an artifact of our different internal languages, because ours didn't have as natural of a mapping to the provided Picobot API. I am unsure if you would be able to get a bit better organization by splitting your code into a few files, like one for this main file, another for partial rules, etc. I can't quite tell from looking at your code.